rs/index/src/collection/collection.rs

```diff
@@ -40,6 +40,7 @@ pub struct Collection<Q: Quantizer + Clone + Send + Sync> {
     versions_info: RwLock<VersionsInfo>,
     base_directory: String,
     mutable_segment: RwLock<MutableSegment>,
+    tmp_mutable_segment: RwLock<MutableSegment>,
```

Comment:
let's not create a mutable segment when not necessary. Maybe an `Option` or something?